crates/bdk/src/wallet/mod.rs

```diff
@@ -427,27 +428,97 @@ impl<D> Wallet<D> {
             .next()
     }
 
-    /// Return a single transactions made and received by the wallet
-    ///
-    /// Optionally fill the [`TransactionDetails::transaction`] field with the raw transaction if
-    /// `include_raw` is `true`.
-    pub fn get_tx(&self, txid: Txid, include_raw: bool) -> Option<TransactionDetails> {
+    /// Returns a `TxOut` known by this wallet corresponding to `outpoint` if it exists in the
+    /// wallet's database whether spent or not.
+    pub fn get_txout(&self, outpoint: &OutPoint) -> Option<TxOut> {
+        self.indexed_graph
+            .graph()
+            .all_txouts()
+            .filter_map(|(op, txo)| {
+                if op.eq(outpoint) {
+                    Some(txo.clone())
+                } else {
+                    None
+                }
+            })
+            .next()
+    }
+
+    /// The total transaction fee amount, sum of input amounts minus sum of output amounts, in sats.
+    /// If the `Wallet` is missing a TxOut for an input returns None.
+    pub fn fee_amount(&self, tx: &Transaction) -> Option<u64> {
+        let txouts: Option<Vec<TxOut>> = tx
+            .input
+            .iter()
+            .map(|txin| self.get_txout(&txin.previous_output))
+            .collect();
+
+        txouts.map(|inputs| {
+            let input_amount: u64 = inputs.iter().map(|i| i.value).sum();
+            let output_amount: u64 = tx.output.iter().map(|o| o.value).sum();
+            input_amount
+                .checked_sub(output_amount)
+                .expect("input amount must be greater than output amount")
+        })
+    }
+
+    /// The transaction's `FeeRate`. If the `Wallet` is missing a `TxOut` for an input returns None.
+    pub fn fee_rate(&self, tx: &Transaction) -> Option<FeeRate> {
+        let fee_amount = self.fee_amount(tx);
+        fee_amount.map(|fee| {
+            let weight = tx.weight();
+            FeeRate::from_wu(fee, weight)
+        })
+    }
+
+    /// Return `TransactionAmounts` for a `Transaction` in relation to the `Wallet` and it's
+    /// descriptors.
+    pub fn sent_and_received(&self, tx: &Transaction) -> TransactionAmounts {
```

Comment:
Can I rename this to `tx_amounts`? the name `sent_and_received` is a bit long an awkward.

Reply:
ACK

Reply:
Personally I prefer `sent_and_received` as I find it more descriptive.